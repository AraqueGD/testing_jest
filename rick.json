{
    "id": 1,
    "name": "Rick",
    "status": "Alive",
    "species": "Human",
    "gender": "Male"
}