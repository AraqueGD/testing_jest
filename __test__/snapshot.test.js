import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe('Es hora de las Instantaneas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    })
    
    test('Siempre Fallara', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }
        expect(user).toMatchSnapshot();
    })

    test('Tenemos una excepciÃ³n dentro del codigo', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Camilo Araque"
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    })
    

})
