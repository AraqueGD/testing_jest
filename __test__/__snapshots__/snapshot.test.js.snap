// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las Instantaneas Siempre Fallara 1`] = `
Object {
  "createAt": 2021-01-28T01:48:32.487Z,
  "id": 19,
}
`;

exports[`Es hora de las Instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las Instantaneas Tenemos una excepci√≥n dentro del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Camilo Araque",
}
`;
